aten/src/ATen/native/ConvolutionMM3d.cpp

```diff
@@ -343,12 +395,27 @@ void slow_conv3d_backward_out_cpu_template(
   const Tensor grad_output_contiguous = grad_output.contiguous();
   grad_input.resize_as_(input);
   TORCH_CHECK(grad_input.is_contiguous(), "grad_input must be contiguous")
-  fgrad_input.resize_as_(finput);
-  TORCH_CHECK(fgrad_input.is_contiguous(), "fgrad_input must be contiguous")
-  fgrad_input.zero_();
 
+  const int64_t dim_planes = 1;
+  const int64_t dim_depth = 2;
+  const int64_t dim_height = 3;
+  const int64_t dim_width = 4;
+  const int64_t n_input_plane = input.size(dim_planes);
+  const int64_t input_depth = input.size(dim_depth);
+  const int64_t input_height = input.size(dim_height);
+  const int64_t input_width = input.size(dim_width);
+  const int64_t output_depth =
+      (input_depth + 2 * pad_depth - kernel_depth) / stride_depth + 1;
+  const int64_t output_height =
+      (input_height + 2 * pad_height - kernel_height) / stride_height + 1;
+  const int64_t output_width =
+      (input_width + 2 * pad_width - kernel_width) / stride_width + 1;
   const int64_t batch_size = input.size(0);
 
+  Tensor fgrad_input = at::empty({batch_size,
+      n_input_plane * kernel_depth * kernel_height * kernel_width,
+      output_depth * output_height * output_width}, input.options());
+  TORCH_CHECK(fgrad_input.is_contiguous(), "fgrad_input must be contiguous")
```

Comment:
this TORCH_CHECK doesn't seem necessary? since we're constructing a tensor with at::empty it should come out contiguous anyways, right?

Reply:
good catch & I agree!